src/tests/unittests/WireTests.cpp

```diff
@@ -436,6 +435,40 @@ TEST_F(WireTests, StructureOfObjectArrayArgument) {
     FlushClient();
 }
 
+// Test that the wire is able to send structures that contain objects
+TEST_F(WireTests, StructureOfStructureArrayArgument) {
+    static constexpr int NUM_BINDINGS = 3;
+    nxtBindGroupBinding bindings[NUM_BINDINGS]{
+        {0, NXT_SHADER_STAGE_BIT_VERTEX, NXT_BINDING_TYPE_SAMPLER, 2},
+        {1, NXT_SHADER_STAGE_BIT_VERTEX, NXT_BINDING_TYPE_SAMPLED_TEXTURE, 1},
+        {2,
+         static_cast<nxtShaderStageBit>(NXT_SHADER_STAGE_BIT_VERTEX |
+                                        NXT_SHADER_STAGE_BIT_FRAGMENT),
+         NXT_BINDING_TYPE_UNIFORM_BUFFER, 1},
+    };
+    nxtBindGroupLayoutDescriptor bglDescriptor;
+    bglDescriptor.numBindings = NUM_BINDINGS;
+    bglDescriptor.bindings = bindings;
+
+    nxtDeviceCreateBindGroupLayout(device, &bglDescriptor);
+    nxtBindGroupLayout apiBgl = api.GetNewBindGroupLayout();
+    EXPECT_CALL(
+        api,
+        DeviceCreateBindGroupLayout(
+            apiDevice, MatchesLambda([bindings](const nxtBindGroupLayoutDescriptor* desc) -> bool {
+                for (int i = 0; i < NUM_BINDINGS; ++i) {
+                    if (memcmp(&desc->bindings[i], &bindings[i], sizeof(nxtBindGroupBinding)) !=
```

Comment:
Sorry this still could fail if the structure has gaps and they contain values that are different.

For example:

```
struct {
  char a;
  int b;
}
```

is actually

```
struct {
  char a;
  char gap[3];
  int b;
}
```

Reply:
Ah, right, done